### PR TITLE
Distinct search parameter support added

### PR DESCRIPTION
**Pull request by Barabasbalazs**
# Pull Request

## Related issue
Fixes https://github.com/meilisearch/meilisearch-java/issues/756

## What does this PR do?
- Added support for using the _distinct_ keyword in the java client. Updated tests and code samples accordingly.

## PR checklist
Please check if your PR fulfills the following requirements:
- [x] Does this PR fix an existing issue, or have you listed the changes applied in the PR description (and why they are needed)?
- [x] Have you read the contributing guidelines?
- [x] Have you made sure that the title is accurate and descriptive of the changes?

Thank you so much for contributing to Meilisearch!

### CODE REVIEW

**Comment by curquiza** [comment]
@Barabasbalazs can you fix the linter test before I review?

**Comment by Barabasbalazs** [comment]
@curquiza fixed the linting issue, also merged in the changes from main

File: .code-samples.meilisearch.yaml [hunk 1]
```diff
@@ -827,3 +827,13 @@ get_similar_post_1:
     .setId("143")
     .setEmbedder("manual");
   client.index("movies").searchSimilarDocuments(query)
+search_parameter_reference_distinct_1: |-
+  SearchRequest searchRequest = SearchRequest.builder().q("QUERY TERMS").distinct("ATTRIBUTE_A").build();
+  client.index("INDEX_NAME").search(searchRequest);
+distinct_attribute_guide_filterable_1: |-
+  Settings settings = new Settings();
+  settings.setFilterableAttributes(new String[] {"language", "title"});
+  client.index("movies").updateSettings(settings);
```
Comment (curquiza) [inline .code-samples.meilisearch.yaml:836]:
```suggestion
distinct_attribute_guide_filterable_1: |-
  Settings settings = new Settings();
  settings.setFilterableAttributes(new String[] {"product_id", "SKU", "url"});
  client.index("products").updateSettings(settings);
```

File: .code-samples.meilisearch.yaml [hunk 2]
```diff
@@ -827,3 +827,13 @@ get_similar_post_1:
...
+distinct_attribute_guide_distinct_parameter_1: |-
+  SearchRequest searchRequest = SearchRequest.builder().q("and").distinct("language").build();
+  client.index("movies").search(searchRequest);
```
Comment (curquiza) [inline .code-samples.meilisearch.yaml:839]:
```suggestion
distinct_attribute_guide_distinct_parameter_1: |-
  SearchRequest searchRequest = SearchRequest.builder().q("white shirt").distinct("sku").build();
  client.index("products").search(searchRequest);
```

**Review by curquiza — APPROVED** [review]
thank you

bors merge

**Pull request merged**